File: NoiseEngine/Logging/LogLevel.cs
using System;

namespace NoiseEngine.Logging;

[Flags]
public enum LogLevel {
    Debug = 1 << 0,
    Trace = 1 << 1,
    Info = 1 << 2,
    Warning = 1 << 3,
    Error = 1 << 4,
    Fatal = 1 << 5,

    None = 0,
    All = Debug | Trace | Info | Warning | Error | Fatal
}

File: NoiseEngine.Native/src/logging/log_level.rs
use log::Level;

#[repr(u8)]
#[derive(Clone, Copy)]
pub enum LogLevel {
    Debug = 1 << 0,
    Trace = 1 << 1,
    Info = 1 << 2,
    Warning = 1 << 3,
    Error = 1 << 4,
}

impl From<Level> for LogLevel {
    fn from(level: Level) -> Self {
        match level {
            Level::Debug => LogLevel::Debug,
            Level::Trace => LogLevel::Trace,
            Level::Info => LogLevel::Info,
            Level::Warn => LogLevel::Warning,
            Level::Error => LogLevel::Error,
        }
    }
}
